--- dwm-5.8.2.orig/dwm.c	2011-01-28 13:46:44.453895251 +0100
+++ dwm-5.8.2.patched/dwm.c	2011-01-28 13:49:27.894005995 +0100
@@ -834,14 +834,16 @@
 
 	if((ev->mode != NotifyNormal || ev->detail == NotifyInferior) && ev->window != root)
 		return;
-	if((m = wintomon(ev->window)) && m != selmon) {
-		unfocus(selmon->sel, True);
-		selmon = m;
+  if (!focusonclick) {
+    if((m = wintomon(ev->window)) && m != selmon) {
+      unfocus(selmon->sel, False);
+      selmon = m;
+    }
+    if((c = wintoclient(ev->window)))
+      focus(c);
+    else
+      focus(NULL);
 	}
-	if((c = wintoclient(ev->window)))
-		focus(c);
-	else
-		focus(NULL);
 }
 
 void
